src/GafferImage/ImageWriter.cpp

```diff
@@ -852,6 +922,15 @@ IECore::MurmurHash ImageWriter::hash( const Context *context ) const
 	IECore::MurmurHash h = TaskNode::hash( context );
 	h.append( fileNamePlug()->hash() );
 	h.append( channelsPlug()->hash() );
+	if( colorSpacePlug()->getValue() == "")
+	{
+		const std::string colorSpaceName = defaultColorSpace();
+		h.append( colorSpaceName );
+	}
+	else
+	{
+		h.append( colorSpacePlug()->getValue() );
```

Comment:
Best to avoid a second call to `colorSpacePlug()->getValue()`, since hash functions are meant to be as past as possible. In fact, maybe we should replace the `defaultColorSpace()` method with a `colorSpace()`, since everywhere we call it we seem to be doing this check for an override anyway.